python.d/neo.chart.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+# Description:  netdata python.d module for monitoring Neo4j
+# Author: Jerome Baton, @wadael , copied from the example of Pawel Krupa (paulfantom)
+
+from bases.FrameworkServices.SimpleService import SimpleService
+
+try:
+    from neo4j.v1 import GraphDatabase
+    NEO4JDRIVER = True
+except ImportError:
+    NEO4JDRIVER = False
+
+
+ORDER = ['monnodes','monlabels','monrels']
+
+CHARTS = {
+    'monnodes': {
+        'options': [None, 'Nodes count', 'Nodes', 'Node count', 'neo4j', 'line'],
+        'lines': [
+             ['id', 'name', 'absolute', 1, 1, 'hidden']
+        ]
+    },
+    'monrels': {
+        'options': [None, 'Relations count', 'Nodes', 'Most present relations', 'neo4j', 'line'],
+        'lines': [
+             ['id', 'name', 'absolute', 1, 1, 'hidden']
+        ]
+    },
+    'monlabels': {
+        'options': [None, 'Relations count', 'Nodes', 'Most present labels', 'neo4j', 'line'],
+        'lines': [
+            ['id', 'name', 'absolute', 1, 1, 'hidden']
+        ]
+    }
+
+
+}
+
+class Service(SimpleService):
+
+    def __init__(self, configuration=None, name=None):
+        SimpleService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self.neosession = None
+        self.alive = True
+
+    def __exit__(self, exc_type, exc_value, traceback):
+       if NEO4JDRIVER:
+            self.neosession.close()
+            self.neodriver.close()
+
+    def check(self):
+        if not NEO4JDRIVER:
+            self.error("the 'neo4j-driver' module is needed to use neo.chart.py. See the script python-modules-installer.sh in /usr/libexec/netdata/python.d")
+            return False
+
+        self.connect()
+        return True
+
+    def get_data_from_cypher_query(self, chartName, data, cypherQuery):
+        try:
+            with self.neosession.begin_transaction() as tx:
+                for neorecord in tx.run(cypherQuery):
+                    if neorecord["label"] not in self.charts[chartName]:
+                        self.charts[chartName].add_dimension([ neorecord["label"] ])
+                    data[ neorecord["label"] ]  = neorecord["cnt"]
+                    tx.close()
+        except Exception:
```

Comment:
make `exception` specific
in case of  `exception` we have to set `self.alive` to `False`

Reply:
@wadael Just a bit of background on the whole `except Exception:` thing since you're reasonably new to Python:

1. This is exactly equivalent to just putting `except:`, so if you _really_ need to catch _everything_ without reporting on what happened, do that instead.
2. Catching everything is almost always a bad idea.  Syntax errors are also a subclass of `Exception`, as are things like passing incorrect variable types to functions (`TypeError`), and trying to access non-existent indexes in iterables or keys in dictionaries (`IndexError` and `KeyError`).  As a result, just catching all types of exceptions can hide real issues with the code itself, which makes debugging much more complicated.  Being specific with what you are actually handling with the except clause is also good practice as it makes sure that things you didn't expect and thus don't properly handle result in actual tracebacks.

In this case, you probably want to specify whatever the Neo4J module's main error class is.